source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.cc

```diff
@@ -19,31 +19,69 @@ namespace StatSinks {
 namespace MetricsService {
 
 GrpcMetricsStreamerImpl::GrpcMetricsStreamerImpl(Grpc::RawAsyncClientSharedPtr raw_async_client,
-                                                 const LocalInfo::LocalInfo& local_info)
+                                                 const LocalInfo::LocalInfo& local_info,
+                                                 uint32_t batch_size)
     : GrpcMetricsStreamer<envoy::service::metrics::v3::StreamMetricsMessage,
                           envoy::service::metrics::v3::StreamMetricsResponse>(raw_async_client),
       local_info_(local_info),
       service_method_(*Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
-          "envoy.service.metrics.v3.MetricsService.StreamMetrics")) {}
+          "envoy.service.metrics.v3.MetricsService.StreamMetrics")),
+      batch_size_(batch_size) {}
 
 void GrpcMetricsStreamerImpl::send(MetricsPtr&& metrics) {
-  envoy::service::metrics::v3::StreamMetricsMessage message;
-  message.mutable_envoy_metrics()->Reserve(metrics->size());
-  message.mutable_envoy_metrics()->MergeFrom(*metrics);
+  bool send_identifier = false;
 
   if (stream_ == nullptr) {
     ENVOY_LOG(debug, "Establishing new gRPC metrics service stream");
     stream_ = client_->start(service_method_, *this, Http::AsyncClient::StreamOptions());
-    // For perf reasons, the identifier is only sent on establishing the stream.
+
+    if (stream_ == nullptr) {
+      ENVOY_LOG(error,
+                "unable to establish metrics service stream. Will retry in the next flush cycle");
+      return;
+    }
+    send_identifier = true;
```

Comment:
can you please put back this comment here 
// For perf reasons, the identifier is only sent on establishing the stream.